libraries/default.rb

```diff
@@ -127,18 +127,19 @@ def private_recipe_ips(cookbook, recipe)
     def resolve_hostname(ip)
       require 'resolv'
       hostf = Resolv::Hosts.new
-      dns = Resolv::DNS.new
-      # Try and resolve hostname first using /etc/hosts, then use DNS
+
       begin
-        hostname = hostf.getname(ip)
+        hname = hostf.getname(ip)
       rescue
         begin
-          hostname = dns.getname(ip)
```

Comment:
Why was this step removed? `elasticsearch-chef` is using this to resolve the IPs of the other elastic nodes. On AWS or anywhere where we don't change the `/etc/hosts` file, this won't work if you have multiple nodes.

Reply:
Also as @kouzant said in a previous comment, you have to cast `dns.getname(ip)` to string

Reply:
ok, i put it back in. elasticsearch uses this function to do a reverse DNS lookup. I know, because it was broken on installations.

Reply:
Here it's wrong. If a recipe on `hopsworks2` asks for the IP of `hopsworks0`, and you don't have the entry in the `/etc/hosts` - then you will get `node['fqdn']` (of the host you are in), in this case `hopsworks2`.